mvebu: fix regulatory country on wrt3200acm

anomeome opened the pull request:
correct oversight on setting regulatory country and mac address for device wrt3200acm wireless configuration: FS#3307

Signed-off-by: Kabuli Chana <newtownBuild@gmail.com>

anomeome committed:
mvebu: add wrt3200acm to 03_wireless CC

correct device CC oversight

Signed-off-by: Kabuli Chana <newtownBuild@gmail.com>

hauke commented:
The change looks good, I think this was always missing.
Could you please improve the commit message, like:
```
mvebu: Fix foo problem on wrt3200acm

Set correct regulatory county and mac address for the wifi radios on wrt3200acm.

Signed-off-by: Kabuli Chana newtownBuild@gmail.com
```

anomeome commented:
Don't think this change should impact MAC setup. Given the mwlwifi readme I question setting the CC to anything on the _rango_ or _venom_, but moot if in agreement with the eeprom.

lucize commented:
the mac from what I see is already working as it should without this change, maybe it would be better to use $labelmac value (ethaddr from uboot) and not the eth0, with eth0 compiled as module the mac is random generated

anomeome commented:
- yes, although seems it should be otherwise
- $label_mac as per 02_network. Not available, but I guess this could change as to method employed there.
- or some other way to get mac

adschm commented:
One doesn't need the label_mac as such, but one could just replace the
`cat /sys/class/net/eth0/address`
by
`mtd_get_mac_ascii devinfo hw_mac_addr`
and adjust the calculation offset if necessary.

anomeome committed:
mvebu: Correct regulatory country of WRT3200ACM

correct oversight on setting regulatory country and mac address of wireless configuration
change method of retrieving mac address

Signed-off-by: Kabuli Chana newtownBuild@gmail.com

anomeome commented:
Tested on a _mamba_ as working.

lucize commented on target/linux/mvebu/cortexa9/base-files/etc/uci-defaults/03_wireless:
```diff
@@ -14,9 +14,10 @@ linksys,wrt1200ac|\
 linksys,wrt32x)
 	SKU=$(strings /dev/mtd3|sed -ne 's/^cert_region=//p')
-	WIFIMAC2G=$(macaddr_add $(cat /sys/class/net/eth0/address) +1)
+	WIFIMAC2G=$(macaddr_add $(mtd_get_mac_ascii devinfo hw_mac_addr) +1)
```
can you test with `ethaddr` instead of `hw_mac_addr` ?
or `mtd_get_mac_ascii u-boot-env ethaddr`
hw_mac_addr will seems to be not available if mvneta is compiled as a module

adschm replied:
@lucize Then this should also be changed in 02_network? (Particularly since board.d happens _before_ uci-defaults ...)

anomeome replied:
Neither of those worked for me on the _mamba_, no error message to be seen. I recall _dengqf6_ having to fight some to get the MAC for this device in the mvebu:DSA PR.

lucize replied:
`fw_printenv | grep ethaddr` is not providing the correct mac whitin os ?

anomeome replied:
It does provide the correct mac.

anomeome commented:
I assume that this patch is correct as is, in terms of resolving the issue of the FS in the OP. If there is anything else that can be tried in support of the PR to reduce kernel size, it will probably also involve 02_network, which will impact other mvebu targets, so probably best left for a subsequent PR.

adschm commented:
We still wait for the commit message update.

anomeome commented:
Alright, sure thought that email was to be automatically be appended.

hauke commented:
Thank you for the patch, it was applied to master.
